Lib/test/test_socket.py

```diff
@@ -1259,6 +1259,38 @@ def testGetServBy(self):
         # Find one service that exists, then check all the related interfaces.
         # I've ordered this by protocols that have both a tcp and udp
         # protocol, at least for modern Linuxes.
+        def query_available_service(*protocols):
+            services_file = '/etc/services'
+            if not os.path.exists(services_file):
+                return None
+            services_found = dict()
+            with open(services_file, 'r') as f:
```

Comment:
This whole logic makes me feel that this could be a public function listing the known services of a given file but that's a separate feature (possibly redundant and not useful).

Reply:
Let's just move `query_available_service` outside (no need for more refactoring, just put your new function at the module-level).